.github/workflows/update_build_status.yml

```diff
@@ -0,0 +1,95 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+name: Update build status workflow
+
+on:
+  schedule:
+  - cron: "*/15 * * * *"
+
+jobs:
+  update:
+    name: Update build status
+    runs-on: ubuntu-20.04
+    steps:
+      - name: "Update build status"
+        uses: actions/github-script@v3
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          script: |
+            const endpoint = 'GET /repos/:owner/:repo/pulls?state=:state'
+            const params = {
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              state: 'open'
+            }
+
+            // See https://docs.github.com/en/graphql/reference/enums#mergestatestatus
+            const maybeReady = ['behind', 'clean', 'draft', 'has_hooks', 'unknown', 'unstable'];
+
+            // Iterate open PRs
+            for await (const prs of github.paginate.iterator(endpoint,params)) {
+              // Each page
+              for await (const pr of prs.data) {
+                console.log('SHA: ' + pr.head.sha)
+                console.log('  Mergeable status: ' + pr.mergeable_state)
+                if (pr.mergeable_state == null || maybeReady.includes(pr.mergeable_state)) {
+                  const checkRuns = await github.request('GET /repos/{owner}/{repo}/commits/{ref}/check-runs', {
+                    owner: context.repo.owner,
+                    repo: context.repo.repo,
+                    ref: pr.head.sha
```

Comment:
If I understand correct, the latest patch still not cover the case mentioned in https://github.com/HyukjinKwon/spark/pull/42#issuecomment-821023453 .

We need do a double check using [Get Pull Request](https://docs.github.com/en/rest/reference/pulls#get-a-pull-request), and check `current_pr.head.sha` == `pr.head.sha`, if not equal we should use `current_pr.head.sha` as the ref input.

Reply:
@Yikun, I think it does handle. The logic here only works if the specific sha has the status check. So as long as the status check has the matched link, it will work. If it doesn't, it will skip.

Reply:
> The logic here only works if the specific sha has the status check.

emm, but old (not latest) commit (multi commits in one PR) still have valid check-runs, I just a little bit confusing how to ensure the commit is the latest?

Reply:
I mean the first commit triggers the first notify check, the second trigger the frist notify check, there are two valid commits with check-runs. and then schedule job happened, the 3rd commit is coming between the schedule job.

Reply:
If this specific pr.head.sha is not matched with the one from motify workflow, it will skip because the pr.head.sha wouldn't have a status.

If pr.head.sha has the status, the sha will always be matched (although it might not be the latest)

Reply:
So as long as a sha has the status check matched with the forked one (which is guaranteed in the notify workflow),  it will keep matched.

Reply:
In the seanrio you mentioned, the third commit will be skipped because it does not have a check run named "Build and test"

Reply:
OK, thanks for the detail explanatation, it make sense, and I also see the schedule job is less than 20s. If it still have the real problem, we will fix it when it happenes.

Reply:
Thank you Yikun for very detailed review:)